internal/encryption/targets.go

```diff
@@ -58,7 +59,21 @@ func (base *baseEncryption) buildTargetMethods(meta map[keyprovider.Addr][]byte)
 	}
 
 	methods, targetDiags := builder.build(base.target, base.name)
-	return methods, append(diags, targetDiags...)
+	diags = append(diags, targetDiags...)
+
+	if base.enforced {
+		for _, m := range methods {
```

Comment:
Hmm, this is a little tricky.  When encrypting, enforced just applies to the first method.  When decrypting it applies to all methods.

Reply:
I agree tofu should check the first one only on encryption. However, I am missing the point with decryption.

Currently, when decrypting it only applies to _a chosen method_ so it's okay to have an `unencrypted` method in the configuration if the state is not plaintext.

But why do we restrict reading plaintext state in the first place? Consistency?

I think I also miss the point of the `enforced` flag. After the migration, users should just remove the `unencrypted` method everywhere and it's a better option than checking `enforced` flag.

Could you please explain a bit? I think we also need to extend the doc with some guidance on the flag.

Reply:
@cam72cam ^

Reply:
In other words:

If a user wants to block the `unencrypted` method from being used on both encryption and decryption, they use `enforced` flag (current implementation, my understanding).

Why not remove the method from the configuration then?

Reply:
> But why do we restrict reading plaintext state in the first place? Consistency?

Security.  If someone can bypass state encryption entirely by replacing the contents with an unencrypted payload, they can inject providers and all sorts of other nasty stuff.

Reply:
@cam72cam yep, that matches my understanding as well. But why would you keep an `unencrypted` method with an `enforced` flag after the migration? It makes a lot more sense to just remove the method completely.

Reply:
So there is a good use-case for this feature when users put their configuration into the env variable. The configuration is being merged and having an enforced flag in the file makes sure the configuration from env is read and valid.

Here is more info: https://github.com/opentofu/opentofu/blob/main/docs/state_encryption.md#environment-configuration

I believe we want to have no `unencrypted` methods in the configuration at all in this scenario so it's okay to just check the whole method list (with no ties to the operation). WDYT?

Reply:
I believe that is valid.